drivers/pinctrl/pinctrl_npcx.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2022 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <assert.h>
+#include <drivers/pinctrl.h>
+#include <soc.h>
+
+/* Driver config */
+struct npcx_pinctrl_config {
+	/* scfg device base address */
+	uintptr_t base_scfg;
+};
+
+static const struct npcx_pinctrl_config npcx_pinctrl_cfg = {
+	.base_scfg = DT_REG_ADDR_BY_NAME(DT_NODELABEL(scfg), scfg),
+};
+
+/* Pin-control local functions for peripheral devices */
+static bool npcx_periph_pinmux_has_lock(int group)
+{
+#if defined(CONFIG_SOC_SERIES_NPCX7)
+	if (group == 0x00 || (group >= 0x02 && group <= 0x04) || group == 0x06 ||
+		group == 0x0b || group == 0x0f) {
+		return true;
+	}
+#elif defined(CONFIG_SOC_SERIES_NPCX9)
+	if (group == 0x00 || (group >= 0x02 && group <= 0x06) || group == 0x0b ||
+		group == 0x0d || (group >= 0x0f && group <= 0x12)) {
+		return true;
+	}
+#endif
+	return false;
+}
+
+static void npcx_periph_pinmux_configure(const struct npcx_periph *alt, bool func,
+	bool is_locked)
+{
+	const uint32_t scfg_base = npcx_pinctrl_cfg.base_scfg;
```

Comment:
why is `uint32_t` used here, and `uintptr_t` used for the `base_scfg`? Please be consistent with types.

Reply:
Done.